GameServer/Logic/Packets/PacketHandlers/Handlers/HandleChatBoxMessage.cs

```diff
@@ -11,10 +11,565 @@
+    unsafe class HandleChatBoxMessage : IPacketHandler
     {
+        static SortedDictionary<string, ChatCommand> ChatCommandsDictionary = new SortedDictionary<string, ChatCommand>();
+        /*{
+            {".ad", new AdCommand(".ad", ".ad bonusAd")},
+            {".ap", new ApCommand(".ap", ".ap bonusAp")},
+            {".ch",  new ChCommand(".ch", ".ch championName")},
+            {".coords",  new CoordsCommand(".coords", "")},
+            {".gold",  new GoldCommand(".gold", ".gold goldAmount")},
+            {".health",  new HealthCommand(".health", ".health maxHealth")},
+            {".help",  new HelpCommand(".help", "")},
+            {".inhib",  new InhibCommand(".inhib", "")},
+            {".junglespawn",  new JunglespawnCommand(".junglespawn", "")},
+            {".level",  new LevelCommand(".level", ".level level")},
+            {".mana",  new ManaCommand(".mana", ".mana maxMana")},
+            {".mobs",  new MobsCommand(".mobs", ".mobs teamNumber")},
+            {".model",  new ModelCommand(".model", ".model modelName")},
+            {".packet",  new PacketCommand(".packet", "No idea, too lazy to read the code")},
+            {".set",  new SetCommand(".set", ".set masterMask fieldMask")},
+            {".size",  new SizeCommand(".size", ".size size")},
+            {".skillpoints",  new SkillpointsCommand(".skillpoints", "") },
+            {".spawn",  new SpawnCommand(".spawn", "")},
+            {".speed",  new SpeedCommand(".speed", ".speed speed")},
+            {".tp",  new TpCommand(".tp", ".tp x y")},
+            {".xp",  new XpCommand(".xp", ".xp xp")}
+        };*/
+
+        abstract class ChatCommand
+        {
+            public string Command { get; set; }
+            public string Syntax { get; set; }
+            public bool IsHidden { get; set; }
+            public bool IsDisabled { get; set; }
+
+            public ChatCommand(string command, string syntax)
+            {
+                if (ChatCommandsDictionary.ContainsKey(command))
+                {
+                    Logger.LogCoreInfo("The command \"" + command + "\"" + " already exists in the chat commands dictionary, won't be added again.");
+                    return;
+                }
+                this.Command = command;
+                this.Syntax = syntax;
+                ChatCommandsDictionary.Add(command, this);
+            }
+
+            public abstract void Execute(ENetPeer* peer, Game game, bool hasReceivedArguments, string arguments = "");
+
+            public void ShowSyntax() {
+                SendDebugMsgFormatted(DebugMsgType.SYNTAX, this.Syntax);
+            }
+        }
+
+        #region Command classes
+        class AdCommand : ChatCommand
+        {
+            public AdCommand(string command, string syntax) : base(command, syntax) { }
+
+            public override unsafe void Execute(ENetPeer* peer, Game game, bool hasReceivedArguments, string arguments = "")
```

Comment:
It's probably needed as `ENetPeer*` is a pointer, which are unsafe in C# as far as I know